### PR TITLE
Review 2023

### PR DESCRIPTION
**Pull request by mrtn78**

### CODE REVIEW

File: ch01_Inleiding.md [hunk 1]
```diff
@@ -73,45 +73,44 @@ Digikoppeling is beschreven in een set van documenten. Deze set is als volgt opg
...
-DigiKoppeling gaat over logistiek, dus over de envelop en niet over de
-inhoud. De hele set info die tezamen nodig is voor een complete generieke
-DigiKoppeling koppelvlakdefinitie (Raamwerk Specificatie genoemd)
-bestaat uit:
-* interfacedefinitie “on the wire”, (voorbeeld)listing van SOAP headers, en
-* informatie over velden en hun specifieke inhoud.
+De koppelvlakspecificatie beschrijft de eisen waar de adapters aan moeten voldoen om interoperabel met elkaar te kunnen communiceren.
+Digikoppeling gaat over logistiek, dus over de envelop en niet over de inhoud. De hele set info die tezamen nodig is voor een complete generieke
+Digikoppeling koppelvlakdefinitie (Raamwerk Specificatie genoemd) bestaat uit:
+* interfacedefinitie (voorbeeld)listing van HTTP eb SOAP headers, en
```
Comment (sanderke) [inline ch01_Inleiding.md]:
```suggestion
* interfacedefinitie (voorbeeld)listing van HTTP en SOAP headers, en
```

File: ch02_Best Practices.md [hunk 2]
```diff
@@ -2,29 +2,36 @@
...
-### Halen
+### Pull principe
 
-Er is bij grote berichten sprake van halen, wanneer een groot bestand door de ontvangende partij wordt opgehaald. Voorafgaand aan deze actie dient altijd een meta-bericht verstuurd te worden door de verzender.
+Er is bij grote berichten sprake van pull, wanneer een groot bestand door de ontvangende partij wordt opgehaald. Voorafgaand aan deze actie dient altijd een meta-bericht verstuurd te worden door de verzender. Dit kan met behulp van het REST-API, WUS of ebMS profiel.
 
 #### EbMS2
 
-Bij het Melden van grote hoeveelheden informatie dient van ebMS2 gebruik gemaakt te worden. Met ebMS2 kan de ontvangst van de metadata gegarandeerd worden.
+Bij het Melden van grote hoeveelheden informatie kan van ebMS2 gebruik gemaakt te worden. Met ebMS2 kan de ontvangst van de metadata gegarandeerd worden.
```
Comment (sanderke) [inline ch02_Best Practices.md]:
```suggestion
Bij het Melden van grote hoeveelheden informatie kan van ebMS2 gebruik gemaakt worden. Met ebMS2 kan de ontvangst van de metadata gegarandeerd worden.
```

File: ch02_Best Practices.md [hunk 3]
```diff
@@ -351,11 +360,11 @@ Belangrijkste foutsituaties bij bestandsoverdracht:
...
+Strik genomen moeten zowel de server als de client controleren op correct OIN gebruik. Dit houdt in dat het OIN uit het certificaat dient overeen te komen met de OIN uit het bericht. In praktijk zal dit voor de client echter niet altijd noodzakelijk zijn. Ook bij het profiel wordt deze controle niet strikt uitgevoerd en bovendien kun je je afvragen waarom de http-client het OIN van de server moet controleren als net daarvoor metadata met een url is ontvangen van een communicatie-partner waarvan het OIN bij de ontvangst van deze meta-data al gecontroleerd is. Bovendien controleert een TLS-client automatisch of het domein van de server overeenkomt met het certificaat.
```
Comment (sanderke) [inline ch02_Best Practices.md]:
```suggestion
Strikt genomen moeten zowel de server als de client controleren op correct OIN gebruik. Dit houdt in dat het OIN uit het certificaat dient overeen te komen met het OIN uit het bericht. In praktijk zal dit voor de client echter niet altijd noodzakelijk zijn. Ook bij het profiel wordt deze controle niet strikt uitgevoerd en bovendien kun je je afvragen waarom de http-client het OIN van de server moet controleren als net daarvoor metadata met een url is ontvangen van een communicatie-partner waarvan het OIN bij de ontvangst van deze meta-data al gecontroleerd is. Bovendien controleert een TLS-client automatisch of het domein van de server overeenkomt met het certificaat.
```

File: ch03_Use cases.md [hunk 4]
```diff
@@ -2,84 +2,88 @@
...
+Dit hoofdstuk beschrijft deze use-cases en geeft aan hoe deze gerealiseerd kunnen worden. Daarbij maakt de uitwerking duidelijk of gegevens via REST-API, ebMS2, WUS en/of als groot bericht verstuurd worden. Beschrijving van de use-cases is generiek; specifieke implementaties kunnen variëren maar zullen vaak deze generieke patronen als basis hebben.
 
 ## Download
 
-In deze use case zijn benodigde gegevens al beschikbaar in een download-server. De afnemer dient echter nog de beschikking te krijgen over een referentie naar de gewenste gegevens. Daartoe worden de volgende processtappen doorlopen:
+In deze use case is benodigde gegevens set al beschikbaar in een download-server bij de service provider. De client dient echter nog de beschikking te krijgen over een verwijzing naar de gewenste gegevens set. Daartoe worden de volgende processtappen doorlopen:
```
Comment (sanderke) [inline ch03_Use cases.md]:
```suggestion
In deze use case is een benodigde gegevensset al beschikbaar in een download-server bij de service provider. De client dient echter nog de beschikking te krijgen over een verwijzing naar de gewenste gegevensset. Daartoe worden de volgende processtappen doorlopen:
```

File: ch03_Use cases.md [hunk 5]
```diff
@@ -2,84 +2,88 @@
...
 
-![Use case 'Download'](media/use_case_download.jpeg "Use case 'Download'")
+![Use case 'Download'](media/use_case_download.svg "Use case 'Download'")
 
-- De Afnemer bevraagt (WUS-request of eventueel ebMS2) de Service Provider met kenmerkende criteria voor de gezochte gegevens. Eventueel kan deze bevraging in enkele tussenstappen verlopen als deze initieel teveel mogelijkheden oplevert.
-
-- De Service provider levert een verwijzing naar de gezochte gegevens in de vorm van Meta-data (WUS-response of ebMS2).
-
-- De Afnemer haalt de gewenste gegevens op (Groot Bericht) en krijgt deze op grond van autorisatie (OIN).
+1. De service requester / client bevraagt (REST-API, WUS of ebMS2 request) de service provider met kenmerkende criteria voor de gezochte gegevens set. Eventueel kan deze bevraging in enkele tussenstappen verlopen als deze initieel teveel mogelijkheden oplevert.
```
Comment (sanderke) [inline ch03_Use cases.md]:
```suggestion
1. De service requester / client bevraagt (REST-API, WUS of ebMS2 request) de service provider met kenmerkende criteria voor de gezochte gegevensset. Eventueel kan deze bevraging in enkele tussenstappen verlopen als deze initieel te veel mogelijkheden oplevert.
```

File: ch03_Use cases.md [hunk 6]
```diff
@@ -2,84 +2,88 @@
...
+2. De service provider levert een verwijzing naar de gezochte gegevens set in de vorm van Meta-data (WUS-response of ebMS2).
+3. De service requester haalt de gewenste gegevens set op (Groot Bericht) en krijgt deze op grond van autorisatie (OIN).
```
Comment (sanderke) [inline ch03_Use cases.md]:
```suggestion
3. De service requester haalt de gewenste gegevensset op (Groot Bericht) en krijgt deze op grond van autorisatie (OIN).
```

File: ch03_Use cases.md [hunk 7]
```diff
@@ -2,84 +2,88 @@
...
-1. De Afnemer bevraagt (ebMS2-request of eventueel WUS) de Service Provider met kenmerkende criteria voor de gezochte gegevens. Eventueel kan deze bevraging in enkele tussenstappen verlopen als deze initieel teveel mogelijkheden oplevert.
-
-1. De Service Provider maakt de gewenste gegevens aan en zet deze klaar in een Groot Bestand.
+In deze use case vraagt een service requester een gegevensselectie van een service provider. Daartoe worden de volgende processtappen doorlopen:
 
-1. Zodra het Grote Bestand gereed is, Meldt de Service Provider dit aan de eerdere Afnemer met een verwijzing naar de gezochte gegevens in de vorm van Meta-data (ebMS2).
+![Use case 'Selectie'](media/use_case_selectie.svg "Use case 'Selectie'")
 
-1. De Afnemer haalt de gewenste gegevens op (Groot Bericht) en krijgt deze op grond van autorisatie (OIN).
+1. De service requester bevraagt (ebMS2-request of eventueel REST-API of WUS) de service provider met kenmerkende criteria voor de gezochte gegevens set. Eventueel kan deze bevraging in enkele tussenstappen verlopen als deze initieel teveel mogelijkheden oplevert.
```
Comment (sanderke) [inline ch03_Use cases.md]:
```suggestion
1. De service requester bevraagt (ebMS2-request of eventueel REST-API of WUS) de service provider met kenmerkende criteria voor de gezochte gegevensset. Eventueel kan deze bevraging in enkele tussenstappen verlopen als deze initieel te veel mogelijkheden oplevert.
```

File: ch03_Use cases.md [hunk 8]
```diff
@@ -2,84 +2,88 @@
...
+1. De service provider maakt de gewenste gegevens set aan en zet deze klaar in een Groot Bestand.
+1. Zodra het Grote Bestand gereed is, Meldt de service provider dit aan de eerdere service requester met een verwijzing naar de gezochte gegevens set in de vorm van Meta-data (ebMS2).
+1. De service requester haalt de gewenste gegevens set op (Groot Bericht) en krijgt deze op grond van autorisatie (OIN).
```
Comment (sanderke) [inline ch03_Use cases.md]:
```suggestion
1. De service requester haalt de gewenste gegevensset op (Groot Bericht) en krijgt deze op grond van autorisatie (OIN).
```

File: ch03_Use cases.md [hunk 9]
```diff
@@ -2,84 +2,88 @@
...
+## Push
 
-1. De Service Provider stuurt een 'verzoek tot ophalen' naar de Afnemer (ebMS2). In deze Melding is in ieder geval de Meta-data van het Grote Bestand opgenomen, maar kan bijvoorbeeld ook informatie over de aard en beoogde afhandeling van het Grote Bestand zijn opgenomen.
+In deze use case verzendt een service provider gegevens naar een service requester. Daartoe worden de volgende processtappen doorlopen:
 
-1. De Afnemer haalt de gegevens op (Groot Bericht) en krijgt deze verstrekt op grond van autorisatie (OIN).
+![Use case 'Push'](media/use_case_push.svg "Use case 'Push'")
 
-Aanvullend kan het Grote Bestand ook verwijderd worden. Bijvoorbeeld nadat de expiration-time is verstreken of nadat de Afnemer een bericht heeft gestuurd om de succesvolle ontvangst te bevestigen.
+Op enig moment is er een gebeurtenis waardoor de service provider besluit tot verzenden van gegevens. Voorbeelden van triggers zijn: tijd, bericht ontvangst of wijziging van gegevensobjecten.
```
Comment (sanderke) [inline ch03_Use cases.md]:
```suggestion
Op enig moment is er een gebeurtenis waardoor de service provider besluit tot verzenden van gegevens. Voorbeelden van triggers zijn: tijd, berichtontvangst of wijziging van gegevensobjecten.
```

File: ch03_Use cases.md [hunk 10]
```diff
@@ -2,84 +2,88 @@
...
+1. De service requester stuurt een 'upload verzoek' naar de service provider. In deze melding is in ieder geval de Meta-data van het Grote Bestand opgenomen, maar kan bijvoorbeeld ook informatie over de aard en beoogde afhandeling van het Grote Bestand zijn opgenomen.
+2. De service provider stuurt een 'upload response' met instructies over de upload lokatie (UUID).
```
Comment (sanderke) [inline ch03_Use cases.md]:
```suggestion
2. De service provider stuurt een 'upload response' met instructies over de uploadlocatie (UUID).
```

File: ch03_Use cases.md [hunk 11]
```diff
@@ -2,84 +2,88 @@
...
 
-In deze use case distribueert een Service-verlener gegevens naar meerdere Afnemers. Daartoe worden de volgende processtappen doorlopen:
-
-![Use case '(multi-) distributie'](media/use_case_multidistributie.jpeg "Use case '(multi-) distributie'")
+In deze use case distribueert een service provider gegevens naar meerdere Afnemers. Daartoe worden de volgende processtappen doorlopen:
 
-1. Op enig moment is er een gebeurtenis waardoor de Service-provider besluit tot distributie van gegevens. Voorbeelden van triggers zijn: tijd, bericht ontvangst of wijziging van gegevensobjecten.
+![Use case '(multi-) distributie'](media/use_case_multidistributie.svg "Use case '(multi-) distributie'")
 
+1. Op enig moment is er een gebeurtenis waardoor de service provider besluit tot distributie van gegevens. Voorbeelden van triggers zijn: tijd, bericht ontvangst of wijziging van gegevensobjecten.
```
Comment (sanderke) [inline ch03_Use cases.md]:
```suggestion
1. Op enig moment is er een gebeurtenis waardoor de service provider besluit tot distributie van gegevens. Voorbeelden van triggers zijn: tijd, berichtontvangst of wijziging van gegevensobjecten.
```

File: ch03_Use cases.md [hunk 12]
```diff
@@ -2,84 +2,88 @@
...
 ## Upload
 
-![Use case 'Upload'](media/use_case_upload.jpeg "Use case 'Upload'")
-
-In deze use case upload een Afnemer gegevens naar een Service-verlener. Deze use case is alleen mogelijk met de 'brengen' interactie die de huidige versie niet ondersteund. De use case “verzenden” is functioneel hiermee gelijkwaardig maar past wel het 'halen' interactiepatroon toe. De processtappen van “upload” zijn daarom niet verder uitgewerkt.
+In deze use case upload een client naar een service provider. Voor het voorbeeld is Digipoort gebruikt als service provider en met behulp van het Push principe wordt met behulp van een REST-API koppeling de metadata gedeeld. Daartoe worden de volgende processtappen doorlopen:
 
+![Use case 'Upload'](media/use_case_upload.svg "Use case 'Upload'")
 
+1. Op enig moment is er een gebeurtenis waardoor de service requester besluit tot verzenden van gegevens. Voorbeelden van triggers zijn: tijd, bericht ontvangst of wijziging van gegevensobjecten.
```
Comment (sanderke) [inline ch03_Use cases.md]:
```suggestion
1. Op enig moment is er een gebeurtenis waardoor de service requester besluit tot verzenden van gegevens. Voorbeelden van triggers zijn: tijd, berichtontvangst of wijziging van gegevensobjecten.
```

File: ch03_Use cases.md [hunk 13]
```diff
@@ -2,84 +2,88 @@
...
+2. Als de gegevens set nog niet beschikbaar is in de vorm van een Groot Bestand zal dit aangemaakt worden (bijvoorbeeld door samenstelling vanuit een database)
```
Comment (sanderke) [inline ch03_Use cases.md]:
```suggestion
2. Als de gegevensset nog niet beschikbaar is in de vorm van een Groot Bestand zal dit aangemaakt worden (bijvoorbeeld door samenstelling vanuit een database)
```

File: js/config.js [hunk 14]
```diff
@@ -4,29 +4,33 @@ var respecConfig = {
...
     ],
   github: "https://github.com/Logius-standaarden/Digikoppeling-Best-Practices-GB",
   previousPublishDate: "2017-10-01",
-  previousPublishVersion: "3.1",
+  previousPublishVersion: "3.1.1",
   pubDomain: "dk",
-  publishDate: "2022-04-11",
-  publishVersion: "3.1.1",
+  publishDate: "2023-05-16",
+  publishVersion: "3.2",
```
Comment (sanderke) [inline js/config.js]:
```suggestion
  publishVersion: "3.2.0",
```
Bij "Documentbeheer" staat `3.2.0`.
https://github.com/Logius-standaarden/Digikoppeling-Best-Practices-GB/blob/90e8e9b99dbe66fd4804e033702dbb1aecb54be2/02_documentbeheer.md?plain=1#L10

**Review by sanderke — CHANGES_REQUESTED** [review]

**Review by mrtn78 — COMMENTED** [review]
typo's fixed

**Review by sanderke — APPROVED** [review]

File: ch03_Use cases.md [hunk 15]
```diff
@@ -2,84 +2,88 @@
...
+In deze use case vraagt een service requester een gegevensselectie van een service provider. Daartoe worden de volgende processtappen doorlopen:
 
-1. Zodra het Grote Bestand gereed is, Meldt de Service Provider dit aan de eerdere Afnemer met een verwijzing naar de gezochte gegevens in de vorm van Meta-data (ebMS2).
+![Use case 'Selectie'](media/use_case_selectie.svg "Use case 'Selectie'")
 
-1. De Afnemer haalt de gewenste gegevens op (Groot Bericht) en krijgt deze op grond van autorisatie (OIN).
+1. De service requester bevraagt (ebMS2-request of eventueel REST-API of WUS) de service provider met kenmerkende criteria voor de gezochte gegevensset. Eventueel kan deze bevraging in enkele tussenstappen verlopen als deze initieel te veel mogelijkheden oplevert.
+1. De service provider maakt de gewenste gegevens set aan en zet deze klaar in een Groot Bestand.
+1. Zodra het Grote Bestand gereed is, Meldt de service provider dit aan de eerdere service requester met een verwijzing naar de gezochte gegevens set in de vorm van Meta-data (ebMS2).
+1. De service requester haalt de gewenste gegevensset op (Groot Bericht) en krijgt deze op grond van autorisatie (OIN).
```
Comment (PHaasnoot) [inline ch03_Use cases.md:26]: opsomming?

File: 03_colofon.md [hunk 16]
```diff
@@ -1,5 +1,6 @@
 # Colofon
 
 | | |
-|---|---|
-| Logius Servicecentrum: | Postbus 96810 <br>2509 JE Den Haag<br>tel. 0900 555 4555 (10 ct p/m)<br>email servicecentrum@logius.nl |
+|------------------------------|---------------------------------------------------------------------------------------------------------|
+| Logius Servicecentrum:       | Postbus 96810 <br>2509 JE Den Haag<br>tel. 0900 555 4555 (10 ct p/m)<br>email. servicecentrum@logius.nl |
+| Logius Afdeling Standaarden: | email. digikoppeling@logiuis.nl<br>email. api@logius.nl                                                 |
```
Comment (PHaasnoot) [inline 03_colofon.md]:
```suggestion
| Logius Afdeling Standaarden: | email. digikoppeling@logius.nl<br>email. api@logius.nl                                                 |
```

**Review by PHaasnoot — APPROVED** [review]

**Pull request merged**